src/node/routes/logout.ts

```diff
@@ -1,17 +1,19 @@
 import { Router } from "express"
-import { getCookieDomain, redirect } from "../http"
-import { Cookie } from "./login"
+import { Cookie, getCookieDomain, redirect } from "../http"
+import { sanitizeString } from "../util"
 
 export const router = Router()
 
-router.get("/", async (req, res) => {
+router.get<{}, undefined, undefined, { base?: string; to?: string }>("/", async (req, res) => {
+  const path = sanitizeString(req.query.base) || "/"
+  const to = sanitizeString(req.query.to) || "/"
+
   // Must use the *identical* properties used to set the cookie.
-  res.clearCookie(Cookie.Key, {
+  res.clearCookie(Cookie.SessionKey, {
     domain: getCookieDomain(req.headers.host || "", req.args["proxy-domain"]),
-    path: req.query.base || "/",
+    path: decodeURIComponent(path),
```

Comment:
This is encoded on the VS Code side:

https://github.com/cdr/vscode/blob/5e0c6f3b95ed032e62c49101ae502a46c62ef202/src/vs/workbench/browser/client.ts#L184-L197

Reply:
I think this path has to be the same as the other cookie in order to overwrite it?  So I think we will need to apply the same logic here we did in the login, namely the whole `path.posix.join()` bit.

I suppose we will want `..` if browsing at `/vscode/` and `.` otherwise.

Reply:
I'm guessing we'll need to test that to verify. If you're correct, I think one of us will have to implement that then @code-asher. Do you want to or do you want me to?

Reply:
Yeah I think if you go to `http://localhost:8080/vscode/` and click "log out" then it will (un)set the cookie at `/vscode/` which will not log you out since the cookie would have been initially set at `/`.

If you get to it before I do feel free but otherwise I will take it on!